rsrc/rsrc.go

```diff
@@ -0,0 +1,107 @@
+package rsrc
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/akavel/rsrc/binutil"
+	"github.com/akavel/rsrc/coff"
+	"github.com/akavel/rsrc/ico"
+	"github.com/akavel/rsrc/internal"
+)
+
+const (
+	RT_ICON       = coff.RT_ICON
+	RT_GROUP_ICON = coff.RT_GROUP_ICON
+	RT_MANIFEST   = coff.RT_MANIFEST
+)
+
+// on storing icons, see: http://blogs.msdn.com/b/oldnewthing/archive/2012/07/20/10331787.aspx
+type GRPICONDIR struct {
+	ico.ICONDIR
+	Entries []GRPICONDIRENTRY
+}
+
+func (group GRPICONDIR) Size() int64 {
+	return int64(binary.Size(group.ICONDIR) + len(group.Entries)*binary.Size(group.Entries[0]))
+}
+
+type GRPICONDIRENTRY struct {
+	ico.IconDirEntryCommon
+	Id uint16
+}
+
+func Embed(fnameout, arch, fnamein, fnameico string) error {
+	newid := make(chan uint16)
+	go func() {
+		for i := uint16(1); ; i++ {
+			newid <- i
+		}
+	}()
```

Comment:
In a library, the above block becomes a resource leak; we must fix this too. 

### Proposal 1 (preferred, but more work in call sites, incl. addicon signature)

```
lastid := uint16(0)
newid := func() uint16 {
    lastid++
    return lastid
}
```

### Proposal 2 (uglier but quicker, in case above fails or you don't have time for more of this)

```
// FIXME(akavel): newid := func() uint16 { lastid++; return lastid; }
newid := ...
quit := make(chan struct{})
defer close(quit)
go func() {
    for ... {
        select {
        case newid <- i:
        case <-quit:
            return
        }
    }
}()
```